src/app/common/hooks/account/use-old-wallet-salt.ts

```diff
@@ -0,0 +1,91 @@
+import { oldWalletSalt } from '@app/store/utils/vault-reducer-migration';
+import { decryptContent, encryptContent, getPublicKeyFromPrivate } from '@stacks/encryption';
+import { createWalletGaiaConfig } from '@stacks/wallet-sdk';
+import { useAsync } from 'react-async-hook';
+import { GaiaHubConfig, uploadToGaiaHub } from '@stacks/storage';
+import { fetchPrivate } from '@stacks/common';
+import { gaiaUrl } from '@shared/constants';
+import { useWalletState } from '@app/store/wallet/wallet.hooks';
+import { logger } from '@shared/logger';
+
+const oldWalletSaltFromGaia = 'old-wallet-salt-from-gaia';
+
+export function useBackupOldWalletSalt() {
```

Comment:
Wouldn't this better take place in background script? 🤔  If it works here, that's okay, but as we only essentially need to do this once, we would be doing less work putting the logic there.

Reply:
Yes that's what I thought afterwards.